submission/submission-core/src/main/kotlin/ac/uk/ebi/biostd/submission/domain/request/SubmissionRequestCleaner.kt

```diff
@@ -1,104 +1,91 @@
-import ebi.ac.uk.extended.model.StorageMode
-import ebi.ac.uk.extended.model.storageMode
-import kotlinx.coroutines.flow.collectIndexed
-import kotlinx.coroutines.flow.filter
-import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.withIndex
 import mu.KotlinLogging
 import uk.ac.ebi.events.service.EventsPublisherService
-import uk.ac.ebi.extended.serialization.service.ExtSerializationService
-import uk.ac.ebi.extended.serialization.service.filesFlow
 
 private val logger = KotlinLogging.logger {}
 
 class SubmissionRequestCleaner(
+    private val concurrency: Int,
+    private val queryService: SubmissionPersistenceQueryService,
     private val storageService: FileStorageService,
-    private val serializationService: ExtSerializationService,
     private val eventsPublisherService: EventsPublisherService,
-    private val queryService: SubmissionPersistenceQueryService,
     private val requestService: SubmissionRequestPersistenceService,
     private val filesRequestService: SubmissionRequestFilesPersistenceService,
 ) {
+    /**
+     *  Executes the clean submission stage where conflicted files (files updated in new submission version) are
+     *  deleting to be able to persist new one.
+     */
     suspend fun cleanCurrentVersion(
         accNo: String,
         version: Int,
         processId: String,
     ) {
-        requestService.onRequest(accNo, version, LOADED, processId, {
-            cleanCurrentVersion(it.submission)
+        requestService.onRequest(accNo, version, INDEXED_CLEANED, processId, {
+            val previousVersion = it.previousVersion
+            if (previousVersion != null) cleanFiles(accNo, version, previousVersion = previousVersion, CONFLICTING)
             RqtUpdate(it.withNewStatus(CLEANED))
         })
         eventsPublisherService.requestCleaned(accNo, version)
     }
 
-    private suspend fun cleanCurrentVersion(new: ExtSubmission) {
-        val current = queryService.findExtByAccNo(new.accNo, includeFileListFiles = true)
-        if (current != null) {
-            logger.info { "${new.accNo} ${new.owner} Started cleaning common files of version ${new.version}" }
-            deleteCommonFiles(new, current)
-            logger.info { "${new.accNo} ${new.owner} Finished cleaning common files of version ${new.version}" }
-        }
+    /**
+     * Executes the finalize or submission processing stage when files deprecated (file not used any more) from previous
+     * version are deleted. Note that submission is query wth negative version as new version has been already
+     * persisted at this point.
+     */
+    suspend fun finalizeRequest(
+        accNo: String,
+        version: Int,
+        processId: String,
+    ) {
+        requestService.onRequest(accNo, version, PERSISTED, processId, {
+            val previousVersion = it.previousVersion
+            if (previousVersion != null) cleanFiles(accNo, version, previousVersion = -previousVersion, DEPRECATED)
```

Comment:
I thought we were storing deprecated files in the current submission request

Reply:
not following this comment. As submission is deprecated and we need to load submission data (ralpath and secret) that is the reason we search for submission with negative version